src/dev/flang/ast/Block.java

```diff
@@ -358,10 +358,19 @@ public Expr propagateExpectedType(Resolution res, AbstractFeature outer, Abstrac
         // something else:
         _expressions.add(new Block(new List<>()));
       }
-    Expr resExpr = removeResultExpression();
+
+    // we must not remove result expression just yet.
+    // we rely on it being present in SourceModule.inScope()
+    var idx = resultExpressionIndex();
+    Expr resExpr = resultExpressionIndex() >= 0
+      ? _expressions.get(idx)
+      : null;
```

Comment:
Can you use `var resExpr = resultExpression()` instead, which is inherted from `AbstractBlock`?